File: Java/src/model/Manager.java
package model;

import java.util.ArrayList;
import java.util.Date;

public class Manager extends Employee {
    private Department dept;
    private int numEmployeesManaging;
    private String currDeptManaging;
    private ArrayList<Employee> employeesManaging = new ArrayList<>();
    
    public Manager(int employeeId, String firstName, String lastName,
                   Date dob, String currentPosition, Department dept) {
        super(employeeId, firstName, lastName, dob, currentPosition, dept);
        this.dept = dept;
        this.currDeptManaging = dept.toString();
    }
    
    @Override
    public void reportToManager() {
        System.out.println("[MANAGER] " + getFirstName() + " " + getLastName() + 
                          " is reporting to upper management. Managing " + 
                          numEmployeesManaging + " employees in " + currDeptManaging);
    }
    
    public void respondToEmployee(Employee employee) {
        System.out.println("Manager " + getFirstName() + " is responding to " + 
                          employee.getFirstName() + " " + employee.getLastName());
    }
    
    public void addEmployee(Employee employee) {
        employeesManaging.add(employee);
        numEmployeesManaging = employeesManaging.size();
        System.out.println("Employee " + employee.getFirstName() + " added to team of manager " + getFirstName());
    }
    
    public Department getManagedDept() { return dept; }
    public int getNumEmployeesManaging() { return numEmployeesManaging; }
}

File: CSharp/Models/Manager.cs
using System;
using System.Collections.Generic;

namespace EmployeeManagementSystem.Models
{
    public class Manager : Employee
    {
        private Department dept;
        private int numEmployeesManaging;
        private string currDeptManaging;
        
        // Collection of employees supervised by this manager.
        // Demonstrates aggregation/association in UML.
        private List<Employee> employeesManaging = new List<Employee>();
        
        // Calls the Employee constructor using base().
        // Demonstrates constructor chaining and inheritance.
        public Manager(int employeeId, string firstName, string lastName,
                      DateTime dob, string currentPosition, Department dept)
            : base(employeeId, firstName, lastName, dob, currentPosition, dept)
        {
            this.dept = dept;
            this.currDeptManaging = dept.ToString();
        }
        
        // Overrides the Employee ReportToManager method.
        // Demonstrates polymorphism and dynamic binding
        public override void ReportToManager()
        {
            Console.WriteLine($"[MANAGER] {FirstName} {LastName} is reporting to upper management. Managing {numEmployeesManaging} employees in {currDeptManaging}");
        }
        
        public void RespondToEmployee(Employee employee)
        {
            Console.WriteLine($"Manager {FirstName} is responding to {employee.FirstName} {employee.LastName}");
        }
        
        public void AddEmployee(Employee employee)
        {
            employeesManaging.Add(employee);
            numEmployeesManaging = employeesManaging.Count;
            Console.WriteLine($"Employee {employee.FirstName} added to team of manager {FirstName}");
        }
        
        // Read-only properties demonstrate encapsulation
        public Department ManagedDept => dept;
        public int NumEmployeesManaging => numEmployeesManaging;
    }
}